### ISSUE TITLE
Overriding Bootstrap

### ISSUE THREAD

**Issue by navidmesbah** [mxstbr/react-boilerplate#620]
Using twitter bootstrap....

Consider an element: 
**`< input className="btn btn-default" >`**

The Old way to customize the button is to add custom classes: 
**`< input className="btn btn-default btn-circle" >`**

```
.btn-circle{
 bla bla bla
}
```

Is there any way to accomplish this, with CSS Modules approach? How?

**Comment by mxstbr** [comment]
Works the same way:

```JS
<input classname={"btn btn-default " + styles.btnCircle} />
```

and in your styling:

```CSS
.btnCircle {
  bla bla bla
}
```

**Comment by navidmesbah** [comment]
thank you very much!

**Comment by sahat** [comment]
@navidmesbah @mxstbr How would you override built-in tags like `<body>`, `<a>`?

**Comment by navidmesbah** [comment]
just edit one of the styles.css files imported in your project (the one with the app container makes sense) and modify the built-in tags or add a custom.css file after the bootstrap on app.js file.

(app.js file):
import '!!style-loader!css-loader!../vendor/css/bootstrap.min.css';
import '!!style-loader!css-loader!styles/css/custom.css';

Update: it is recommended to use this literal for mixing classNames:

 <input className={`btn btn-default   ${styles.btnCircle}`} />

**Comment by sahat** [comment]
I hope there is a cleaner way to do it. My problem is that Bootstrap CSS is overriding my custom global CSS inside `containers/App/styles.css`.

<img width="1033" alt="screenshot 2016-09-17 22 02 16" src="https://cloud.githubusercontent.com/assets/544954/18612997/b68f2846-7d22-11e6-9f74-9d8cc9be5174.png">

**Comment by paothegit** [comment]
I have the exactly same issue, tried to modify webpack configuration, but no luck.
Only on production build, sanitize.css and bootstrap.css comes after my custom global css file.
Is there a proper way to sort the order of styles? (like `chunkSortMode` in HtmlWebpackPlugin)

Thanks!

**Comment by sahat** [comment]
@paothegit Yes, I ended up importing Bootstrap CSS in the *app/app.js*:

```js
import 'bootstrap/dist/css/bootstrap.css';
```

Then moving "App styles" to the App component, where I think it belongs:

```js
// before - app.js
import styles from 'containers/App/styles.css';

// after - components/App.js
import './styles.css';
```

And then use `:global` to override it, e.g:

```css
:global .navbar-inverse .navbar-nav > li > a {
  color: #252525;
  font-size: 12px;
  font-weight: 500;
  text-transform: uppercase;
  transition: all 0.2s linear;
}
```

Works great so far. 👍

**Comment by paothegit** [comment]
@sahat 
Is `bootstrap/dist/css/bootstrap.css` in your code same as `bootstrap-css-only` which is suggested by this boilerplate?

**Comment by sahat** [comment]
@paothegit I am using the `bootstrap` package, not `bootstrap-css-only` so that's why path is slightly different. Yes I followed those steps initially.

You are right, it actually doesn't work in production build because of how module styles are imported before Bootstrap. I'll see if I can change the order, if not, then using Sass and importing bootstrap directly into App's *style.scss* could potentially work.

**Comment by paothegit** [comment]
@sahat 
I see, I'm trying less version of bootstrap, still finding way to make it working on production build.
Will post here once I succeed.
Thanks!

**Comment by sahat** [comment]
I got it working. Basically, you want to extract bootstrap.css from compiled main.js into compiled main.css.

Add to **prod** webpack config:

```js
vendorCssLoaders: ExtractTextPlugin.extract(
    'style-loader',
    'css-loader'
),
```

Add to **dev** webpack config:

```js
vendorCssLoaders: 'style-loader!css-loader',
```

Update **base** webpack config:
```js
{
        // Do not transform vendor's CSS with CSS-modules
        // The point is that they remain in global scope.
        // Since we require these CSS files in our JS or CSS files,
        // they will be a part of our compilation either way.
        // So, no need for ExtractTextPlugin here.
        test: /\.css$/,
        include: /node_modules/,
        loader: options.vendorCssLoaders,
},
```

**Comment by micheleb** [comment]
sahat's comment fixed issue #1102 for me, so thanks!

Since `ExtractTextPlugin` was complaining, I had to change the first snippet a little:

    vendorCssLoaders: ExtractTextPlugin.extract({
        fallbackLoader: 'style-loader',
        loader: 'css-loader',
    }),
